test/utils.ts

```diff
@@ -164,3 +166,45 @@ export const SYNTAX_CASES = [
 export const testCompiled = process.env.TEST_COMPILED === '1'
 
 export const srcDir = testCompiled ? 'lib' : 'src'
+
+type RuleTesterOptions = {
+  parser?: string
+} & Omit<TSESLint.RuleTesterConfig, 'parser'>
+
+export class RuleTester extends TSESLint.RuleTester {
+  constructor(option: RuleTesterOptions = {}) {
+    /**
+     * The parser option is required in TSESLint.RuleTester, but not in ESLint.RuleTester.
+     * Since TSESLint.RuleTester is just an alias of ESLint.RuleTester, we can safely ignore the type error.
+     */
+    // @ts-expect-error -- parser is not required
+    super(option)
+  }
+
+  /**
+   * Since we can't override the original `run` method, we have to create a new one that accepts our custom rule module with custom meta.
+   */
+  run$<MessageIds extends string, Options extends Readonly<unknown[]>>(
+    ruleName: string,
+    rule: RuleModuleWithCustomMeta<MessageIds, Options>,
+    tests: TSESLint.RunTests<MessageIds, Options>,
+  ) {
+    const { meta, ...ruleWithoutMeta } = rule
+    const { docs, ...metaWithoutDocs } = meta
+    const {
+      category: _1,
+      recommended: _2,
+      ...docsWithoutCategoryAndRecommend
+    } = docs
+
+    const modifiedRule: Parameters<TSESLint.RuleTester['run']>[1] = {
+      ...ruleWithoutMeta,
+      meta: {
+        ...metaWithoutDocs,
+        docs: docsWithoutCategoryAndRecommend,
+      },
+    }
+
+    return super.run(ruleName, modifiedRule, tests)
```

Comment:
Under the hood, `run$` does some extra modification before `super.run`.

Reply:
Is that possible to use `// @ts-expect-error` here instead of create a new `run$` method?

Reply:
@JounQin 

Adding a `ts-expect-error` and moving on sounds very tempting, but not a best practice here.

TypeScript throws a type mismatch for a reason: [Inheritance implies subtyping](https://stackoverflow.com/questions/23592131/what-is-the-difference-between-subtyping-and-inheritance-in-oo-programming), and subtype mismatch breaks the Liskov substitution. So TypeScript disallows inheriting from a base class while not being a valid subtype of it.

Reply:
Hmm... I think it's some kind of balance choice here, or we can say that the typings in `tseslint` is incorrect? The modifications are quite huge, aren't they? I'm thinking about the maintenance vs the patch one.

Reply:
Sorry, this is quite far over my head. Only feedback I would have is `run$` looks a bit dirty to me. I'd suggest some camelCase name instead.